src/structured_config.py

```diff
@@ -111,6 +112,19 @@ def log_cleaner_delete_retention_ms_validator(cls, value: str) -> Optional[int]:
             return int_value
         raise ValueError("Value out of range.")
 
+    @validator("ssl_principal_mapping_rules")
+    @classmethod
+    def ssl_principal_mapping_rules_validator(cls, value: str) -> Optional[str]:
+        """Check that the list is formed by valid regex values."""
+        # get all regex up until replacement position "/"
+        pat = re.compile(r"RULE:([^/]+)(?:,RULE:[^/]+)*(?:DEFAULT){0,1}")
+        matches = re.findall(pat, value)
+        for match in matches:
+            try:
+                re.compile(match)
+            except re.error:
+                raise ValueError("Non valid regex pattern")
+
```

Comment:
I believe we should return the value if everything works

`return value`

Reply:
yeah good catch, I was scratching my head not understanding why unit tests didn't pass :D